scipy/signal/tests/test_waveforms.py

```diff
@@ -314,5 +316,34 @@ def test_integer_tpr(self):
         assert_equal(int_result, float_result, err_msg=err_msg)
 
 
+class TestUnitImpulse(TestCase):
+
+    def test_no_index(self):
+        assert_array_equal(waveforms.unit_impulse(7), [1, 0, 0, 0, 0, 0, 0])
+        assert_array_equal(waveforms.unit_impulse((3, 3)),
+                           [[1, 0, 0], [0, 0, 0], [0, 0, 0]])
+
+    def test_index(self):
+        assert_array_equal(waveforms.unit_impulse(10, 3),
+                           [0, 0, 0, 1, 0, 0, 0, 0, 0, 0])
+        assert_array_equal(waveforms.unit_impulse((3, 3), (1, 1)),
+                           [[0, 0, 0], [0, 1, 0], [0, 0, 0]])
+
+    def test_mid(self):
+        assert_array_equal(waveforms.unit_impulse((3, 3), 'mid'),
+                           [[0, 0, 0], [0, 1, 0], [0, 0, 0]])
+        assert_array_equal(waveforms.unit_impulse(9, 'mid'),
+                           [0, 0, 0, 0, 1, 0, 0, 0, 0])
+
+    def test_dtype(self):
+        imp = waveforms.unit_impulse(7)
+        assert np.issubdtype(imp.dtype, np.float)
```

Comment:
I think we use `assert_true` here (nose-style) instead of plain `assert ` statements (pytest-style)